### DISCUSSION TITLE
How to optimize BlasBatchSize

### DISCUSSION THREAD

**Discussion by maddes8cht** [LostRuins/koboldcpp#110] [Q&A]
The `-blasbatchsize` argument seems to be set automatically if you don't specify it explicitly.
How do I find the optimal setting for this? Does anyone have more Info on the `--blasbatchsize` argument?
With my RTX 3060 (12 GB) and `--useclblast 0 0` I actually feel well equipped, but the performance gain is disappointingly minimal.
Can I improve this with a better setting of `--blasbatchsize`? Or are there settings that can improve something?

**Comment by LostRuins** [comment]
Mostly trial and error, for me I have good results with 256 and 512. Anything smaller or larger and it slows down.

Make sure it is using the correct GPU `--useclblast 0 0` will show the name of the GPU when it triggers, make sure it matches.

**Comment by maddes8cht** [comment]
Thank you.
Yes, `--useclblast 0 0` finds my RTX 3060.
I see the Cuda option is still in development. At the moment, the performance gain is disappointingly small, only slightly measurable and not noticeable. I'm curious if this will improve significantly in the near future.

**Reply by LostRuins** [reply]
CUDA option is a struggle, because windows makes it very hard to build for it if you are not using visual studio. We are hoping to accelerate CLBlast fast enough that it wont be needed.
